src/test/java/hu/dungeonhunter/TestFight.java

```diff
@@ -169,397 +179,302 @@ public void monsterCallerGoblinTest() {
-        champion.setAccuracy(1000);
-        fight.setChampion(champion);
-        fight.setChampionFinalInitiation(1000);
-
-        fight.battle();
-
-        ifAssertThatIsChampionStillAlive();
-        softly.assertThat(fight.getMonsterFactory().getKilledMonsterCounter()).isEqualTo(startingKilledMonsterCounter + 1);
-        softly.assertThat(fight.getMonsterCounter()).isEqualTo(startingMonsterCounter - 1);
-        softly.assertThat(fight.getChampion().getHealingPotionCounter()).isEqualTo(startingHealingPotionCounter + 1);
-        ifAssertThatIsChampionStillAlive(); //this is the new goblin
-        softly.assertAll();
-    }
-
-    @Test
-    public void battleChampionWithHighInitiationAndMonsterWithHighAccuracyAndDefenseTest() {
-        Champion champion = new Champion();
+    public void championCriticalMissEvent1Test() {
         champion.setHp(1);
         fight.setChampion(champion);
-        fight.setChampionFinalInitiation(1000);
-        MonstersInterface monster = monsterFactory.getMonster(CharacterTypes.GOBLIN);
-        monster.setAccuracy(1000);
-        monster.setDefense(1000);
+        Goblin monster = new Goblin();
         fight.setMonster(monster);
+        fight.setRandomCriticalMissEvent(1);
+        int[] championHpAfterDamage = {fight.getChampion().getHp() - fight.getChampion().getMaxDamage(),
+                fight.getChampion().getHp() - fight.getChampion().getNumOfDices()};
 
-        fight.battle();
+        fight.criticalMissEvent(champion);
 
+        softly.assertThat(fight.getChampion().getHp()).isBetween(championHpAfterDamage[0], championHpAfterDamage[1]);
         ifAssertThatIsChampionDefeated();
-        ifAssertThatIsMonsterStillAlive();
-        softly.assertThat(fight.getMonsterFactory().getKilledMonsterCounter()).isEqualTo(startingKilledMonsterCounter);
-        softly.assertThat(fight.getMonsterCounter()).isEqualTo(startingMonsterCounter);
-        softly.assertThat(fight.getChampion().getHealingPotionCounter()).isEqualTo(startingHealingPotionCounter);
-        softly.assertThat(fight.nextTurn()).isFalse();
         softly.assertAll();
     }
 
     @Test
-    public void battleChampionWithHighInitiationAndHighDefenseAndMonsterWithHighDefenseTest() {
-        Champion champion = new Champion();
-        champion.setHp(1);
-        champion.setDefense(1000);
-        fight.setChampion(champion);
-        fight.setChampionFinalInitiation(1000);
-        MonstersInterface monster = monsterFactory.getMonster(CharacterTypes.GOBLIN);
-        monster.setDefense(1000);
+    public void monsterCriticalMissEvent1Test() {
+        Goblin monster = new Goblin();
+        monster.setHp(1);
         fight.setMonster(monster);
+        fight.setRandomCriticalMissEvent(1);
+        int[] monsterHpAfterDamage = {fight.getMonster().getHp() - fight.getMonster().getMaxDamage(),
+                fight.getMonster().getHp() - fight.getMonster().getNumOfDices()};
```

Comment:
I believe this is incorrect, as you don't count in the number of dices to the may damage.
```suggestion
        int[] monsterHpAfterDamage = {fight.getMonster().getHp() - fight.getMonster().getMaxDamage() * fight.getMonster().getNumOfDices(),
                fight.getMonster().getHp() - fight.getMonster().getNumOfDices()};
```

Reply:
Its true! I fixed this fail for all these types of  tests.